crates/rpc/src/server/api.rs

```diff
@@ -208,6 +211,18 @@ impl api_server::Api for RpcService {
         let tx = ProvenTransaction::read_from_bytes(&request.transaction)
             .map_err(|err| Status::invalid_argument(format!("Invalid transaction: {err}")))?;
 
+        // Only allow deployment transactions for new network accounts
+        if tx.account_id().is_network() {
+            match tx.account_update().details() {
+                AccountUpdateDetails::New(_) => {},
```

Comment:
It feels like we should have something like `AccountUpdateDetails::is_new_account()` (let's add it in `miden-base`?). But also, could we not use `matches!` macro here for brevity?